sample-notificator-domain/src/test/groovy/life/qbic/business/notification/refactor/SendEmailSpec.groovy

```diff
@@ -0,0 +1,69 @@
+package life.qbic.business.notification.refactor
+
+import life.qbic.business.notification.create.NotificationContent
+import spock.lang.Specification
+
+/**
+ * <p>Tests for the logic of when and how to send notification emails.</p>
+ */
+class SendEmailSpec extends Specification {
+
+  EmailGenerator emailGenerator = Stub()
+  EmailSender flawlessEmailSender = Mock()
+  FailureEmailSender adminInformer = Mock()
```

Comment:
Why cant the admin informer not also be an emailsender? but he could just have another type of emails, no NotificationContent but something else?

Reply:
The adminInformer is an email sender `FailureEmailSender`. As we need the method that this interface provides it needed to be more specific.

Reply:
Originally, this was the idea. I introduced a NotificationEmail and a FailureEmail. As it turned out the process of constructing and sending those emails was quite different and they did not have anything in common except being called email. Thus I simplified the code which made the FailureEmail not being used any longer and allowed for different interfaces with more explicit methods.
I would like to keep this as is.

Reply:
Sure, I just wanted to understand